src/main/java/buildcraft/factory/BlockTank.java

```diff
@@ -88,123 +88,210 @@ public IIcon getIconAbsolute(IBlockAccess iblockaccess, int i, int j, int k, int
-                            if (liquid.amount > 0) {
-                                int qty = tank.fill(ForgeDirection.UNKNOWN, liquid, false);
-                                tank.fill(ForgeDirection.UNKNOWN, container.drain(current, qty, true), true);
-                            }
+            // tank -> container logic
+            FluidStack available = clicked.getTankInfo(ForgeDirection.UNKNOWN)[0].fluid;
+            if (available != null) {
+                ItemStack preview = FluidContainerRegistry.fillFluidContainer(available, held);
+                FluidStack willContain = FluidContainerRegistry.getFluidForFilledItem(preview);
+                if (willContain != null) {
+                    if (!BuildCraftCore.debugWorldgen && !creative) {
+                        if (held.stackSize > 1) {
+                            if (!player.inventory.addItemStackToInventory(preview)) return false;
+                            player.inventory
+                                    .setInventorySlotContents(player.inventory.currentItem, InvUtils.consumeItem(held));
+                        } else {
+                            player.inventory
+                                    .setInventorySlotContents(player.inventory.currentItem, InvUtils.consumeItem(held));
+                            player.inventory.setInventorySlotContents(player.inventory.currentItem, preview);
                         }
                     }
-
+                    clicked.drain(ForgeDirection.UNKNOWN, willContain.amount, true);
+                    syncPlayerAndTanks(world, x, y, z, player, clicked);
                     return true;
                 }
             }
-        } else if (DEBUG_MODE) {
-            TileEntity tile = world.getTileEntity(i, j, k);
-
-            if (tile instanceof TileTank) {
-                TileTank tank = (TileTank) tile;
-                if (tank.getTankInfo(ForgeDirection.UNKNOWN)[0].fluid != null) {
-                    entityplayer.addChatComponentMessage(
-                            new ChatComponentText(
-                                    "Amount: " + tank.getTankInfo(ForgeDirection.UNKNOWN)[0].fluid.amount + " mB"));
+            return false;
+        }
+
+        // IFluidContainerItem (IFCI) Logic (Large Fluid Cells)
+        if (!(held.getItem() instanceof IFluidContainerItem)) return false;
+        if (world.isRemote) return true; // server does transfers
+
+        final IFluidContainerItem fluidContainer = (IFluidContainerItem) held.getItem();
+        final boolean sneaking = player.isSneaking();
+        final int INF_MB = Integer.MAX_VALUE / 4;
+
+        // Support for stacked cells
+        ItemStack work = held.copy();
+        work.stackSize = 1;
+
+        // prioritizes the bottom for tank stacks
+        TileTank bottom = clicked.getBottomTank();
+        if (bottom == null) return true;
+
+        int moved = 0;
+
+        // Push from cell -> tank
+        if (!sneaking) {
```

Comment:
instead of playing with sneaking in one of my certus quartz prs I just disallowed to interact with IFluidContainer items that are also placeable blocks which are mainly tanks:  
https://github.com/GTNewHorizons/AE2FluidCraft-Rework/pull/358

what do you think about this behavior? not sure that you often need to move fluid from one kind of tank to another, and if we're talking about the same kind of tank then placing it on top will make the fluid flow down automatically